metric-schema-java/src/main/java/com/palantir/metric/schema/UtilityGenerator.java

```diff
@@ -564,18 +574,19 @@ private static void generateMetricFactoryBuilder(
                     // TODO(ckozak): Update to use a method which can log a warning and replace existing gauges.
                     // See MetricRegistries.registerWithReplacement.
                     .addStatement(
-                            "$L.$L($L(), $L)",
+                            "$L.$L($N(), $L)",
```

Comment:
I think these should all have been `$N` because they're name references, shouldn't make much difference in practice outside of cases like this where we pass a *Spec. Up to you if you'd like to replace them here or not.

Reply:
Oh interesting, `$N` also [supports](https://github.com/square/javapoet/blob/ba4395e8dae03738a1768d7f6ee02e78fbf44faf/src/main/java/com/squareup/javapoet/CodeBlock.java#L339) `CharSequence`. I assumed it was just for `Spec` types because the [docs](https://github.com/square/javapoet/tree/master#n-for-names) say "Use `$N` to refer to another generated declaration by its name."